### ISSUE TITLE
Sharp damages image integrity

### ISSUE THREAD

**Issue by JamesJGoodwin** [lovell/sharp#2327]
Are you using the latest version? Is the version currently in use as reported by `npm ls sharp` the same as the latest version as reported by `npm view sharp dist-tags.latest`?
**Yes**
What are the steps to reproduce?
**Open an image with `fs.readFile` and then open the same image with `sharp('image.jpeg')` or `const buff = fs.readFile(file); sharp(buff)` and compare buffers or their base64's or md5 hashes of base64's.**
What is the expected behaviour?
**Image buffer should be the same as if you opened the file via `fs.readFile`**
Are you able to provide a minimal, standalone code sample, without other dependencies, that demonstrates this problem?
**Yes, it's 1 question above.**
Are you able to provide a sample image that helps explain the problem?
**Please, refer to https://repl.it/@JamesJGoodwin/FirsthandClumsyBlocks to be able to download an image and run a minimal, standalone code sample**
What is the output of running `npx envinfo --binaries --system`?
```
System:
    OS: Windows 10 10.0.19041
    CPU: (4) x64 Intel(R) Core(TM) i3-4160 CPU @ 3.60GHz
    Memory: 1.83 GB / 7.94 GB
  Binaries:
    Node: 14.6.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.6 - C:\Program Files\nodejs\npm.CMD
```

**Comment by lovell** [comment]
> console.log(`Source opened with sharp: ${md5(await sharp('image.jpg').toString('base64'))}`)
> console.log(`Source opened with sharp second time: ${md5(await sharp('image.jpg').toString('base64'))}`)
> console.log(`Source buffer provided to sharp: ${md5(await sharp(imageBuff).toString('base64'))}`)

> Source opened with sharp: 1441a7909c087dbbe7ce59881b9df8b9
> Source opened with sharp second time: 1441a7909c087dbbe7ce59881b9df8b9
> Source buffer provided to sharp: 1441a7909c087dbbe7ce59881b9df8b9

`toString()` is not part of the sharp API but is instead part of the global JavaScript `Object` API - see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString

The sample code is testing the ability to generate an md5 hash of the string "[object Object]".

```sh
$ node -p "require('md5')('[object Object]')"
1441a7909c087dbbe7ce59881b9df8b9
```

**Comment by JamesJGoodwin** [comment]
> toString() is not part of the sharp API but is instead part of the global JavaScript Object API

Yeah, I know that. But the problem is not in base64 encoding, but in `Buffer` that is returned by sharp. I took an image and saved it with `fs.writeFile` and with `sharp(sourceImageBuffer).toFile('sharp.jpg')` and opened them in hexdump and that's where I discovered that they have different content. Still visually images are the same.

![image](https://user-images.githubusercontent.com/17240507/90341854-f4876f00-e00b-11ea-90f8-cf0d34699132.png)

I'm not sure if this is a problem to have hex differences, but obviously this is not an expected behaviour when you don't make any changes to the image, yet still it's buffer value is modified. Even if you provide sharp with corrent buffer and ask it to give the buffer back (`sharp(sourceImageBuffer).toBuffer()`) - the buffer will be different.
